### PR TITLE
Fix 186 noisy raffling

### LINKED ISSUE #186: Make raffling less noisy
We give an error log for every failed raffle. Only give an error if all raffling attempts fail, not just one.

### PR DESCRIPTION
**Pull request by rbval**
Closes #186

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -76,7 +76,7 @@ services:
       - 5001:5001
   
   minio:
-    image: minio/minio
+    image: minio/minio:9423-d242680
```
Comment (gaieges) [inline docker-compose.yml:79]: do we need to lock this?

File: hmt_escrow/job.py [hunk 2]
```diff
@@ -1432,9 +1436,10 @@ def _create_escrow(self,
                 escrow_created = True
                 break
             except Exception as e:
-                LOG.error(
-                    f"Contract creation failed with {gas_payer} and {gas_payer_priv} due to {e}. Raffling new ones..."
-                )
+                if gas_payer == self.multi_credentials[-1][0]:
+                    LOG.error(
```
Comment (gaieges) [inline hmt_escrow/job.py:1440]: can you make this a `LOG.exception`?

File: hmt_escrow/job.py [hunk 3]
```diff
@@ -1432,9 +1436,10 @@ def _create_escrow(self,
                 escrow_created = True
                 break
             except Exception as e:
-                LOG.error(
-                    f"Contract creation failed with {gas_payer} and {gas_payer_priv} due to {e}. Raffling new ones..."
-                )
+                if gas_payer == self.multi_credentials[-1][0]:
```
Comment (gaieges) [inline hmt_escrow/job.py:1439]: perhaps here you should just use an iterator in the loop?  I fear this check might be a bit fragile

File: docker-compose.yml (hunk 1 above)
Comment (rbval) [reply docker-compose.yml:79]: yeah, didn't work with latest, used from recoracle

File: hmt_escrow/job.py (hunk 2 above)
Comment (rbval) [reply hmt_escrow/job.py:1440]: sure

File: hmt_escrow/job.py (hunk 3 above)
Comment (rbval) [reply hmt_escrow/job.py:1439]: yea, good idea

File: docker-compose.yml (hunk 1 above)
Comment (gaieges) [reply docker-compose.yml:79]: Ahh, is there a better pinned version that we can use? Like v0.3 or something?  Looking through their I just see stuff like `minio/minio:RELEASE.2019-02-06T21-16-36Z` which is basically the same as what you're doing there (build specific vs minor version specific)

File: hmt_escrow/job.py (hunk 2 above)
Comment (gaieges) [reply hmt_escrow/job.py]: Hmm one other thing I was thinking here was - is there any action to take aside from the logging here? Ie do we cancel the contract? Or just not create it at all and throw exception?

Comment (rbval) [reply hmt_escrow/job.py]: We'll retry it the next iteration if this time it fails as the status of the contract won't change

File: docker-compose.yml (hunk 1 above)
Comment (rbval) [reply docker-compose.yml:79]: I just took this version from recoracle, I guess we could use something else

File: hmt_escrow/job.py (hunk 2 above)
Comment (rbval) [reply hmt_escrow/job.py]: Cancelling is probably too aggressive.

File: hmt_escrow/job.py (hunk 3 above)
Comment (rbval) [reply hmt_escrow/job.py]: going to use already defined boolean fields to perform this, more elegant.

File: docker-compose.yml (hunk 1 above)
Comment (rbval) [reply docker-compose.yml]: Now it's pinned to the latest release of minio that works.

Comment (gaieges) [reply docker-compose.yml]: Cool yeah no worries, probably about the same as what you had, they generally have better floating version tags (not in minio's case it looks like)

**Pull request merged**